runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/SdkComponents.java

```diff
@@ -54,9 +54,11 @@
 
   private final Set<String> reservedIds = new HashSet<>();
 
+  private String defaultEnvironmentId;
```

Comment:
I'd remove this and handle this directly in `getOnlyEnvironmentId()` below.

Reply:
The information only available in `getOnlyEnvironmentId()` is insufficient for providing a default environment id. I couldn't come up with any better idea. Can you elaborate more on how to handle this directly in the method?

Reply:
I didn't see that you need the PortablePipelineOptions to populate this field. Looks good then.

Reply:
The field could be made final but it's not super important.